### ISSUE TITLE
Server Scanner

### ISSUE THREAD

**Issue by jaswsinc** [websharks/s2member-kb#133]
---
##### KB Article Creation Checklist

- [x] Write initial draft for this KB Article; label this issue `draft` and either `questions` or `tutorials`
- [x] Add [required YAML configuration](https://github.com/websharks/team/wiki/KB-Articles-::-Required-YAML-Configuration)
- [x] Add Tags for this KB Article to the YAML config (see [YAML Keys (Explained)](https://github.com/websharks/team/wiki/KB-Articles-::-Required-YAML-Configuration#yaml-keys-explained))
- [x] Edit and finalize draft for publishing (remove `draft` label, add `draft-finalized` label)
- [x] Assign Issue to yourself and [create Markdown file](https://github.com/websharks/team/wiki/KB-Articles-::-Creating-the-Markdown-File) (remove `draft-finalized` label, add `pending`)
- [x] Project Lead: Review and Publish KB Article (remove `pending` label, add `published` label)

##### Additional TODOs

_Additional TODO list items go here._

---

**Markdown File:** http://wsharks.com/1FgqfHG
**Published KB Article:** http://s2member.com/kb-article/server-scanner/

---

**Comment by ryanaoneil** [comment]
The link on the article doesn't work. http://s2member.com/r/s2-server-scanner/

**Comment by raamdev** [comment]
> The link on the article doesn't work. http://s2member.com/r/s2-server-scanner/

@ryanaoneil I just tested that link and it's working for me.

**Comment by ryanaoneil** [comment]
I'm getting this: *503 (Invalid Key):* Sorry, your access to this file has
expired. Please contact Support for assistance.

from this url:
https://www.s2member.com/?s2member_file_download=ccaps%2Faccess-s2member-ccap-pro%2Fs2-server-scanner.zip&s2member_file_download_key=a00e81f5499e993385c03e9275699866
ᐧ

On Tue, May 26, 2015 at 11:01 AM, Raam Dev <notifications@github.com> wrote:

> The link on the article doesn't work.
> http://s2member.com/r/s2-server-scanner/
>
> @ryanaoneil <https://github.com/ryanaoneil> I just tested that link and
> it's working for me.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/websharks/s2member-kb/issues/133#issuecomment-105577206>
> .
>



-- 
Ryan O'Neil
LOGOSIdentity.com
We get the Word out.

**Comment by raamdev** [comment]
@ryanaoneil It seems that your browser is loading a cached version of that redirect. The domain changed from `www.s2member.com` to just `s2member.com`. This is where http://s2member.com/r/s2-server-scanner/ is pointing now:

http://s2member.com/?s2member_file_download=ccaps%2Faccess-s2member-ccap-pro%2Fs2-server-scanner.zip&s2member_file_download_key=a00e81f5499e993385c03e9275699866

If you clear your browser cache, the redirect should work as expected.

**Comment by ryanaoneil** [comment]
Nice! So, I had https every on and I think that was the issue. It was
forcing https and that somehow pull in the www subdomain. It still pulled
up the invalid key deal but it downloaded.

Thanks for the help!

On Tue, May 26, 2015 at 2:03 PM, Raam Dev <notifications@github.com> wrote:

> @ryanaoneil <https://github.com/ryanaoneil> It seems that your browser is
> loading a cached version of that redirect. The domain changed from
> www.s2member.com to just s2member.com. This is where
> http://s2member.com/r/s2-server-scanner/ is pointing now:
>
>
> http://s2member.com/?s2member_file_download=ccaps%2Faccess-s2member-ccap-pro%2Fs2-server-scanner.zip&s2member_file_download_key=a00e81f5499e993385c03e9275699866
>
> If you clear your browser cache, the redirect should work as expected.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/websharks/s2member-kb/issues/133#issuecomment-105636016>
> .
>



-- 
Ryan O'Neil
LOGOSIdentity.com
We get the Word out.

**Comment by wpfan1000** [comment]
Getting 503 (Invalid Key): Sorry, your access to this file has expired. Please contact Support for assistance.
Looks like it has been an issue for 4 months - please fix.

**Comment by raamdev** [comment]
@wpfan1000 I just tested this link again and the download is working fine: http://s2member.com/r/s2-server-scanner/

**Comment by wpfan1000** [comment]
Hi, thanks for checking.
This is the message I got when i clicked on the link:
503 (Invalid Key): Sorry, your access to this file has expired. Please contact Support for assistance.

**Comment by wpfan1000** [comment]
Just because it works for your doenst mean it will work for everyone else. Maybe it is working for you because you are logged in or have certain priviledges? I think even if it works for you you should modify it because this is the first time in 20 years of using the web that I get this errror message.

**Comment by wpfan1000** [comment]
Here is the same issue that seems to be an issue with your plugin:
https://wordpress.org/support/topic/error-message-and-links-not-working

**Comment by raamdev** [comment]
@wpfan1000 Does this link work?

http://s2member.com/?s2member_file_download=ccaps%2Faccess-s2member-ccap-pro%2Fs2-server-scanner.zip&s2member_file_download_key=a00e81f5499e993385c03e9275699866

(That's the same link that http://s2member.com/r/s2-server-scanner/ redirects to, but if the above link works, that means your browser must have cached an old version of the full link and is attempting to visit the old link.)

**Comment by wpfan1000** [comment]
Still getting - 
503 (Invalid Key): Sorry, your access to this file has expired. Please contact Support for assistance.

**Comment by wpfan1000** [comment]
When I Google the error:
https://www.google.ca/#q=503+%28Invalid+Key%29:+Sorry%2C+your+access+to+this+file+has+expired.+Please+contact+Support+for+assistance.
I see that this error is specific to your plugin, and a few other WP plugins, but not browsers, not Apache servers, or any servers, etc which means to me it is a setting within your plugin.
Also the wording of the error is not a typical browser/server/mysql etc error - it is specific to your plugin and some other sites that may be using your plugin.
This is supposed to be a simple, open, unprotected download link much like any other eg download link to sample software, pdfs.
So if I dont mean to be blunt but if me and others have this issue in the past, present and hence maybe into the future, it is time to fix it permanently.

**Comment by raamdev** [comment]
> Have you tried simply pasting the link into a normal WP page so that there is no permission check around it?

Yes, I tried visiting the link while completely logged out, in a clean Incognito window in Chrome, and it works just fine for me. I also tested it from multiple ISPs. 

---

@jaswsinc Any feedback here?

**Comment by wpfan1000** [comment]
But your link contains a key - s2member_file_download_key=a00e81f5499e993385c03e9275699866
and the error messages says invalid key - surely that tells you:
1) You are attaching a key to the download link which means it is NOT an open, unprotected link
2) Some software is checking this key and saying it is invalid

**Comment by jaswsinc** [comment]
This link works just fine for me:

http://s2member.com/?s2member_file_download=ccaps%2Faccess-s2member-ccap-pro%2Fs2-server-scanner.zip&s2member_file_download_key=a00e81f5499e993385c03e9275699866

It works when:

- I click it.
- If I open it from an incognito window.
- If I run it through a test tool like: http://httpstatus.io/
- If I look at it funny and talk dirty.

---

Regarding the key: `s2member_file_download_key=a00e81f5499e993385c03e9275699866`
This is a universal key; i.e., allows for public access by anyone.

Many browsers will cache redirects in ways that make it hard to clear though. So if you initially stumbled onto the www variation, that could be the reason for failure. https://www.s2member.com/?s2member_file_download=ccaps%2Faccess-s2member-ccap-pro%2Fs2-server-scanner.zip&s2member_file_download_key=a00e81f5499e993385c03e9275699866

See also: http://salferrarello.com/chrome-clear-redirect-cache/

**Comment by kristineds** [comment]
**Request: Update zip file in KB article to include extension in PHP7**

---

From @jaswsinc in this [support ticket #14157](https://websharks.zendesk.com/agent/tickets/14157):

> The extension in PHP7 is called mysqli and I've had the server scanner updated to reflect the proper extension.  Latest version available here: https://raw.githubusercontent.com/websharks/s2member/dev/s2-server-scanner.php

**Comment by raamdev** [comment]
It would also be nice if this article at least listed the basic minimum requirements, along with a link to the [WordPress requirements page](https://wordpress.org/about/requirements/).

**Comment by BillBB** [comment]
I'm not sure how I do the last step "Now open your web browser and visit: /s2-server-scanner.php for a full report."

When I try to visit I get the results of a search.  What am I missing?

**Comment by raamdev** [comment]
@BillBB You need to enter the URL to where you uploaded the file. So, if you uploaded it to the root of your site, you'd want to visit `http://example.com/s2-server-scanner.php` (replace `example.com` with your actual domain).

I went ahead and updated the KB Article to clarify this.
